### ISSUE TITLE
Docker Plugin Reporting values of 0 for CPU and Mem

### ISSUE THREAD

**Issue by sd-charris** [influxdata/telegraf#2976]
## Bug report
Telegraf Docker Plugin is reporting values of 0 for metric groups "docker_container_cpu" and "docker_container_mem".  The "docker" metric seems to be reporting container count and other metrics just fine.

### Relevant telegraf.conf:

```
[[inputs.docker]]
  ## Docker Endpoint
  ##   To use TCP, set endpoint = "tcp://[ip]:[port]"
  ##   To use environment variables (ie, docker-machine), set endpoint = "ENV"
  endpoint = "npipe:////./pipe/docker_engine"

  ## Timeout for docker list, info, and stats commands
  timeout = "5s"

  ## Whether to report for each container per-device blkio (8:0, 8:1...) and
  ## network (eth0, eth1, ...) stats or not
  perdevice = true

  ## Whether to report for each container total blkio and network stats or not
  total = false
```

### System info:

Amazon AMI: Windows_Server-2016-English-Full-Containers-2017.06.14 (ami-3c4b4145)
Windows Server 2016
telegraf-1.3.2_windows_amd64.zip
telegraf-1.3.3_windows_amd64.zip

### Steps to reproduce:

1. Run Container: docker run -d -p 8000:8000 microsoft/aspnet
2. Simulate Load: docker exec -i [containerId] powershell get-childitem -r
3. C:\"Program Files"\Telegraf\telegraf.exe --config C:\"Program Files"\Telegraf\telegraf.conf --test

### Expected behavior:

Memory and CPU usage should be reported similarly to "docker stats"

### Actual behavior:
Memory and CPU usage is reported to be "0i" see below:

>Plugin: inputs.docker, Collection 1
> docker,engine_host=EC2AMAZ-U8A604O,host=EC2AMAZ-U8A604O n_containers_running=1i,n_containers_stopped=0i,n_containers_paused=0i,n_listener_events=0i,n_cpus=8i,n_used_file_descript
ors=-1i,n_goroutines=28i,n_containers=1i,n_images=1i 1498689859000000000
> docker,engine_host=EC2AMAZ-U8A604O,host=EC2AMAZ-U8A604O,unit=bytes memory_total=34359328768i 1498689859000000000
> docker_container_mem,engine_host=EC2AMAZ-U8A604O,container_name=ecstatic_swanson,container_image=microsoft/aspnet,container_version=unknown,host=EC2AMAZ-U8A604O total_rss=0i,rss_
huge=0i,hierarchical_memory_limit=0i,total_pgpgin=0i,mapped_file=0i,total_inactive_file=0i,active_anon=0i,total_rss_huge=0i,usage=0i,total_unevictable=0i,cache=0i,unevictable=0i,pg
majfault=0i,total_writeback=0i,total_inactive_anon=0i,total_pgfault=0i,max_usage=0i,total_pgmafault=0i,total_active_anon=0i,total_cache=0i,active_file=0i,pgfault=0i,limit=0i,writeb
ack=0i,total_mapped_file=0i,inactive_anon=0i,fail_count=0i,pgpgout=0i,inactive_file=0i,total_active_file=0i,total_pgpgout=0i,usage_percent=0,container_id="00c34d44c830f55ba7370eba8
a6ffd17ae12cda974e5c16b741de3b85f924a6a",rss=0i,pgpgin=0i 1498689861000000000
> docker_container_cpu,container_image=microsoft/aspnet,container_version=unknown,host=EC2AMAZ-U8A604O,engine_host=EC2AMAZ-U8A604O,cpu=cpu-total,container_name=ecstatic_swanson usa
ge_in_usermode=36406250i,usage_in_kernelmode=36406250i,throttling_throttled_periods=0i,throttling_throttled_time=0i,container_id="00c34d44c830f55ba7370eba8a6ffd17ae12cda974e5c16b74
1de3b85f924a6a",usage_total=55937500i,usage_system=0i,throttling_periods=0i,usage_percent=0 1498689861000000000
> docker_container_net,host=EC2AMAZ-U8A604O,engine_host=EC2AMAZ-U8A604O,container_name=ecstatic_swanson,container_image=microsoft/aspnet,container_version=unknown,network=5a7b567d-
80d9-4640-8571-a3c57830bf58 rx_dropped=0i,rx_bytes=7158i,rx_errors=0i,tx_packets=62i,rx_packets=28i,tx_bytes=9656i,tx_dropped=0i,tx_errors=0i,container_id="00c34d44c830f55ba7370eba
8a6ffd17ae12cda974e5c16b741de3b85f924a6a" 1498689861000000000

**Comment by kitt1987** [comment]
I have the same problem here and my docker version is **17.03.1-ee-3.** I opened the TCP endpoint of Docker daemon and fetched container stats via HTTP API. The result is below.
```
{"read":"2017-07-14T05:50:12.2688821Z","preread":"0001-01-01T00:00:00Z","pids_stats":{},"blkio_stats":{"io_service_bytes_recursive":null,"io_serviced_recursive":null,"io_queue_recursive":null,"io_service_time_recursive":null,"io_wait_time_recursive":null,"io_merged_recursive":null,"io_time_recursive":null,"sectors_recursive":null},"num_procs":4,"storage_stats":{"read_count_normalized":10153,"read_size_bytes":79970816,"write_count_normalized":35196,"write_size_bytes":227833856},"cpu_stats":{"cpu_usage":{"total_usage":945625000,"usage_in_kernelmode":411250000,"usage_in_usermode":411250000},"throttling_data":{"periods":0,"throttled_periods":0,"throttled_time":0}},"precpu_stats":{"cpu_usage":{"total_usage":0,"usage_in_kernelmode":0,"usage_in_usermode":0},"throttling_data":{"periods":0,"throttled_periods":0,"throttled_time":0}},"memory_stats":{"commitbytes":122474496,"commitpeakbytes":278777856,"privateworkingset":76922880},"name":"/k8s_buildpackaspnet.7fa94573_buildpackaspnet-2603109986-b253g_f8865f46709214b249b458a72d64460e6_c4e8943a-62be-11e7-b148-0669fa643bf9_70e6e9df","id":"796005ec174a345a585c6c5ea8e7af5423e33cf8380db51ff7db3463d8636648","networks":{"5248997c-6079-42b9-8c43-8124c487c4b1":{"rx_bytes":56673121,"rx_packets":544329,"rx_errors":0,"rx_dropped":0,"tx_bytes":40491000,"tx_packets":411492,"tx_errors":0,"tx_dropped":0}}}
{"read":"2017-07-14T05:50:13.2688615Z","preread":"2017-07-14T05:50:12.2688821Z","pids_stats":{},"blkio_stats":{"io_service_bytes_recursive":null,"io_serviced_recursive":null,"io_queue_recursive":null,"io_service_time_recursive":null,"io_wait_time_recursive":null,"io_merged_recursive":null,"io_time_recursive":null,"sectors_recursive":null},"num_procs":4,"storage_stats":{"read_count_normalized":10153,"read_size_bytes":79970816,"write_count_normalized":35196,"write_size_bytes":227833856},"cpu_stats":{"cpu_usage":{"total_usage":945625000,"usage_in_kernelmode":411250000,"usage_in_usermode":411250000},"throttling_data":{"periods":0,"throttled_periods":0,"throttled_time":0}},"precpu_stats":{"cpu_usage":{"total_usage":945625000,"usage_in_kernelmode":411250000,"usage_in_usermode":411250000},"throttling_data":{"periods":0,"throttled_periods":0,"throttled_time":0}},"memory_stats":{"commitbytes":122474496,"commitpeakbytes":278777856,"privateworkingset":76922880},"name":"/k8s_buildpackaspnet.7fa94573_buildpackaspnet-2603109986-b253g_f8865f46709214b249b458a72d64460e6_c4e8943a-62be-11e7-b148-0669fa643bf9_70e6e9df","id":"796005ec174a345a585c6c5ea8e7af5423e33cf8380db51ff7db3463d8636648","networks":{"5248997c-6079-42b9-8c43-8124c487c4b1":{"rx_bytes":56673349,"rx_packets":544333,"rx_errors":0,"rx_dropped":0,"tx_bytes":40491174,"tx_packets":411495,"tx_errors":0,"tx_dropped":0}}}
```
The memory stats is very different from which docker plugin needs. The true memory usage on Windows is the value of `memory_stats.privateworkingset` and the plugin does not use it.
This [code](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/docker/docker.go#L395) shows which memory fields are used by the plugin.

**Comment by danielnelson** [comment]
I believe this was fixed by #3043.
